clients/unshaded/src/main/java/tachyon/client/FileOutStream.java

```diff
@@ -152,8 +148,8 @@ public void flush() throws IOException {
   }
 
   private void getNextBlock() throws IOException {
-    if (mCurrentBlockId != -1) {
-      if (mCurrentBlockLeftByte != 0) {
+    if (mCurrentBlockOutStream != null) {
+      if (mCurrentBlockOutStream.getRemainingSpaceByte() > 0) {
```

Comment:
I think these two conditions can be in the same if() block and will probably be easier to read.